### ISSUE TITLE
IO Error while updating Anaconda2

### ISSUE THREAD

**Issue by ItsAnkit** [conda/conda#4459]
I have installed Anaconda 2 64-bit in windows 8.1 and when i am trying to update conda using,

```
C:\Windows\System32>conda update --all
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment C:\Users\sardwal\Anaconda2:

The following NEW packages will be INSTALLED:

    chardet:            2.3.0-py27_0
    conda-verify:       2.0.0-py27_0
    isort:              4.2.5-py27_0
    numpydoc:           0.6.0-py27_0
    scandir:            1.4-py27_0
    subprocess32:       3.2.7-py27_0

The following packages will be UPDATED:

    anaconda-clean:     1.0.0-py27_0       --> 1.1.0-py27_0
    anaconda-client:    1.5.1-py27_0       --> 1.6.0-py27_0
    anaconda-navigator: 1.3.1-py27_0       --> 1.3.2-py27_0
    astroid:            1.4.7-py27_0       --> 1.4.9-py27_0
    astropy:            1.2.1-np111py27_0  --> 1.3-np111py27_0
    backports_abc:      0.4-py27_0         --> 0.5-py27_0
    beautifulsoup4:     4.5.1-py27_0       --> 4.5.3-py27_0
    bokeh:              0.12.2-py27_0      --> 0.12.4-py27_0
    boto:               2.42.0-py27_0      --> 2.45.0-py27_0
    bottleneck:         1.1.0-np111py27_0  --> 1.2.0-np111py27_0
    cffi:               1.7.0-py27_0       --> 1.9.1-py27_0
    click:              6.6-py27_0         --> 6.7-py27_0
    cloudpickle:        0.2.1-py27_0       --> 0.2.2-py27_0
    conda-build:        2.0.2-py27_0       --> 2.1.2-py27_0
    contextlib2:        0.5.3-py27_0       --> 0.5.4-py27_0
    cryptography:       1.5-py27_0         --> 1.7.1-py27_0
    curl:               7.49.0-vc9_0       [vc9] --> 7.52.1-vc9_0       [vc9]
    cython:             0.24.1-py27_0      --> 0.25.2-py27_0
    cytoolz:            0.8.0-py27_0       --> 0.8.2-py27_0
    dask:               0.11.0-py27_0      --> 0.13.0-py27_0
    datashape:          0.5.2-py27_0       --> 0.5.4-py27_0
    decorator:          4.0.10-py27_0      --> 4.0.11-py27_0
    docutils:           0.12-py27_2        --> 0.13.1-py27_0
    filelock:           2.0.6-py27_0       --> 2.0.7-py27_0
    flask-cors:         2.1.2-py27_0       --> 3.0.2-py27_0
    freetype:           2.5.5-vc9_1        [vc9] --> 2.5.5-vc9_2        [vc9]
    gevent:             1.1.2-py27_0       --> 1.2.1-py27_0
    greenlet:           0.4.10-py27_0      --> 0.4.11-py27_0
    idna:               2.1-py27_0         --> 2.2-py27_0
    ipaddress:          1.0.16-py27_0      --> 1.0.18-py27_0
    ipykernel:          4.5.0-py27_0       --> 4.5.2-py27_0
    ipywidgets:         5.2.2-py27_0       --> 5.2.2-py27_1
    jdcal:              1.2-py27_1         --> 1.3-py27_0
    jinja2:             2.8-py27_1         --> 2.9.4-py27_0
    jpeg:               8d-vc9_2           [vc9] --> 9b-vc9_0           [vc9]
    jupyter_core:       4.2.0-py27_0       --> 4.2.1-py27_0
    lazy-object-proxy:  1.2.1-py27_0       --> 1.2.2-py27_0
    libpng:             1.6.22-vc9_0       [vc9] --> 1.6.27-vc9_0       [vc9]
    libtiff:            4.0.6-vc9_2        [vc9] --> 4.0.6-vc9_3        [vc9]
    llvmlite:           0.13.0-py27_0      --> 0.15.0-py27_0
    lxml:               3.6.4-py27_0       --> 3.7.2-py27_0
    matplotlib:         1.5.3-np111py27_0  --> 2.0.0-np111py27_0
    menuinst:           1.4.1-py27_0       --> 1.4.4-py27_0
    mkl:                11.3.3-1           --> 2017.0.1-0
    mkl-service:        1.1.2-py27_2       --> 1.1.2-py27_3
    multipledispatch:   0.4.8-py27_0       --> 0.4.9-py27_0
    nbformat:           4.1.0-py27_0       --> 4.2.0-py27_0
    nltk:               3.2.1-py27_0       --> 3.2.2-py27_0
    notebook:           4.2.3-py27_0       --> 4.3.1-py27_1
    numba:              0.28.1-np111py27_0 --> 0.30.1-np111py27_0
    numexpr:            2.6.1-np111py27_0  --> 2.6.1-np111py27_2
    numpy:              1.11.1-py27_1      --> 1.11.3-py27_0
    openpyxl:           2.3.2-py27_0       --> 2.4.1-py27_0
    openssl:            1.0.2j-vc9_0       [vc9] --> 1.0.2k-vc9_0       [vc9]
    pandas:             0.18.1-np111py27_0 --> 0.19.2-np111py27_1
    partd:              0.3.6-py27_0       --> 0.3.7-py27_0
    path.py:            8.2.1-py27_0       --> 10.0-py27_0
    pathlib2:           2.1.0-py27_0       --> 2.2.0-py27_0
    pillow:             3.3.1-py27_0       --> 4.0.0-py27_0
    pip:                8.1.2-py27_0       --> 9.0.1-py27_1
    pkginfo:            1.3.2-py27_0       --> 1.4.1-py27_0
    prompt_toolkit:     1.0.3-py27_0       --> 1.0.9-py27_0
    psutil:             4.3.1-py27_0       --> 5.0.1-py27_0
    pycparser:          2.14-py27_1        --> 2.17-py27_0
    pycrypto:           2.6.1-py27_4       --> 2.6.1-py27_5
    pycurl:             7.43.0-py27_0      --> 7.43.0-py27_2
    pyflakes:           1.3.0-py27_0       --> 1.5.0-py27_0
    pylint:             1.5.4-py27_1       --> 1.6.4-py27_1
    pyqt:               5.6.0-py27_0       --> 5.6.0-py27_2
    pytest:             2.9.2-py27_0       --> 3.0.5-py27_0
    python-dateutil:    2.5.3-py27_0       --> 2.6.0-py27_0
    pytz:               2016.6.1-py27_0    --> 2016.10-py27_0
    pywin32:            220-py27_1         --> 220-py27_2
    pyzmq:              15.4.0-py27_0      --> 16.0.2-py27_0
    qt:                 5.6.0-vc9_0        [vc9] --> 5.6.2-vc9_3        [vc9]
    qtawesome:          0.3.3-py27_0       --> 0.4.3-py27_0
    qtpy:               1.1.2-py27_0       --> 1.2.1-py27_0
    ruamel_yaml:        0.11.14-py27_0     --> 0.11.14-py27_1
    scikit-learn:       0.17.1-np111py27_1 --> 0.18.1-np111py27_1
    scipy:              0.18.1-np111py27_0 --> 0.18.1-np111py27_1
    sphinx:             1.4.6-py27_0       --> 1.5.1-py27_0
    spyder:             3.0.0-py27_0       --> 3.1.2-py27_0
    sqlalchemy:         1.0.13-py27_0      --> 1.1.5-py27_0
    toolz:              0.8.0-py27_0       --> 0.8.2-py27_0
    tornado:            4.4.1-py27_0       --> 4.4.2-py27_0
    traitlets:          4.3.0-py27_0       --> 4.3.1-py27_0
    werkzeug:           0.11.11-py27_0     --> 0.11.15-py27_0
    wrapt:              1.10.6-py27_0      --> 1.10.8-py27_0
    xlsxwriter:         0.9.3-py27_0       --> 0.9.6-py27_0
    xlwings:            0.10.0-py27_0      --> 0.10.2-py27_0
    xlwt:               1.1.2-py27_0       --> 1.2.0-py27_0

Proceed ([y]/n)? y

An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : win-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : 2.0.2
         python version : 2.7.12.final.0
       requests version : 2.9.1
       root environment : C:\Users\sardwal\Anaconda2  (writable)
    default environment : C:\Users\sardwal\Anaconda2
       envs directories : C:\Users\sardwal\Anaconda2\envs
          package cache : C:\Users\sardwal\Anaconda2\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : None
           offline mode : False
             user-agent : conda/4.3.8 requests/2.9.1 CPython/2.7.12 Windows/8.1
Windows/6.3.9600



`$ C:\Users\sardwal\Anaconda2\Scripts\conda-script.py update --all`




    Traceback (most recent call last):
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\exceptions.py", l
ine 617, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\cli\main.py", lin
e 137, in _main
        exit_code = args.func(args, p)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\cli\main_update.p
y", line 65, in execute
        install(args, parser, 'update')
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\cli\install.py",
line 347, in install
        execute_actions(actions, index, verbose=not context.quiet)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\plan.py", line 83
7, in execute_actions
        execute_instructions(plan, index, verbose)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\instructions.py",
 line 258, in execute_instructions
        cmd(state, arg)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\instructions.py",
 line 118, in UNLINKLINKTRANSACTION_CMD
        txn = UnlinkLinkTransaction.create_from_dists(index, prefix, unlink_dist
s, link_dists)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\core\link.py", li
ne 121, in create_from_dists
        for dist, pkg_dir in zip(link_dists, pkg_dirs_to_link))
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\core\link.py", li
ne 121, in <genexpr>
        for dist, pkg_dir in zip(link_dists, pkg_dirs_to_link))
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\gateways\disk\rea
d.py", line 71, in read_package_info
        index_json_record = read_index_json(extracted_package_directory)
      File "C:\Users\sardwal\Anaconda2\lib\site-packages\conda\gateways\disk\rea
d.py", line 94, in read_index_json
        with open(join(extracted_package_directory, 'info', 'index.json')) as fi
:
    IOError: [Errno 2] No such file or directory: u'C:\\Users\\sardwal\\Anaconda
2\\pkgs\\numpy-1.11.3-py27_0\\info\\index.json'
```

Please tell me how to resolve this error.....

**Comment by kalefranz** [comment]
You're on conda 4.3.8, which means you're a good conda citizen and keeping the version up-to-date :)

Now, why would numpy not have an `info/index.json` file?  I honestly don't understand just yet.  Can you look now and tell me if the file

    C:\Users\sardwal\Anaconda2\pkgs\numpy-1.11.3-py27_0\info\index.json

currently exists?

**Comment by ItsAnkit** [comment]
Nope, it's not there...

**Comment by kalefranz** [comment]
Do me a favor.  Try `conda clean --packages` and then try again.  And let me know what happens...

**Comment by ItsAnkit** [comment]
I have tried this and it's still giving same error.

**Comment by kalefranz** [comment]
The root cause of this issue was resolved in a 4.3 patch release.  If you're still having it, manually removing the `C:\Users\sardwal\Anaconda2\pkgs\numpy-1.11.3-py27_0` directory should fix the problem.

**Comment by varunrao1991** [comment]
In windows run as conda promt as admin. and type conda update --prefix C:\ProgramData\Anaconda3 anaconda
